charger/sungrow.go

```diff
@@ -242,11 +255,39 @@ func (wb *Sungrow) Phases1p3p(phases int) error {
 		u = 1
 	}
 
-	_, err := wb.conn.WriteSingleRegister(sgRegPhaseSwitch, u)
+	enabled, err := wb.Enabled()
+	if err == nil && enabled {
+		err = wb.Enable(false)
```

Comment:
If err = …

Reply:
Wie meinst du?

Reply:
Erhm. Nvm. Verklickt!

Reply:
Err = und if Statement kannst du in einer Zeile zusammen fassen

Reply:
Achso, du meinst so, oder?

```suggestion
		if err = wb.Enable(false); err != nil {
			return err
		}
```

Stimmt ;)